### PR TITLE
Add documentation for supported architectures in shoot

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area documentation
/kind task

**What this PR does / why we need it**:
This PR adds documentation for supported architecture in shoot worker pool.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/backlog/issues/19

/cc @rfranzke 

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by shafeeqes** [comment]
Please add this to `docs/README.md` as well.

File: docs/usage/shoot_supported_architectures.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Shoot Supported Architectures
+
+Users can create shoot with worker groups having virtual machines of different architectures. CPU architecture of each worker pool can be specified in shoot yaml as follows:-
```
Comment (shafeeqes) [inline docs/usage/shoot_supported_architectures.md]:
```suggestion
Users can create shoot clusters with worker groups having virtual machines of different architectures. CPU architecture of each worker pool can be specified in the `Shoot` specification as follows:-
```

~Not very sure about the capitalization though.~
edited by @rfranzke

File: docs/usage/shoot_supported_architectures.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Shoot Supported Architectures
```
Comment (rfranzke) [inline docs/usage/shoot_supported_architectures.md]:
```suggestion
# Supported CPU Architectures for Shoot Worker Nodes
```

File: docs/usage/shoot_supported_architectures.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+```yaml
+spec:
+  provider:
+    workers:
+    - name: cpu-worker
+      machine:
+        architecture: <some-cpu-architecture>
+```
+
+If there is no value specified for architecture it defaults to `amd64`.
```
Comment (rfranzke) [inline docs/usage/shoot_supported_architectures.md]: You could mention that the architectures must be present in the `CloudProfile`.

**Review by rfranzke — CHANGES_REQUESTED** [review]

File: docs/usage/shoot_supported_architectures.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+```yaml
+spec:
+  provider:
+    workers:
+    - name: cpu-worker
+      machine:
+        architecture: <some-cpu-architecture>
+```
+
+If no value is specified for the architecture field it defaults to `amd64`. For a valid shoot object, a machine should be present in the respective CloudProfile with the same CPU architecture as specified in Shoot yaml.
```
Comment (shafeeqes) [inline docs/usage/shoot_supported_architectures.md]:
```suggestion
If no value is specified for the architecture field, it defaults to `amd64`. For a valid shoot object, a machine should be present in the respective CloudProfile with the same CPU architecture as specified in the `Shoot` yaml.
```

File: docs/usage/shoot_supported_architectures.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+```yaml
+spec:
+  provider:
+    workers:
+    - name: cpu-worker
+      machine:
+        architecture: <some-cpu-architecture>
+```
+
+If no value is specified for the architecture field, it defaults to `amd64`. For a valid shoot object, a machine should be present in the respective CloudProfile with the same CPU architecture as specified in the `Shoot` yaml.
```
Comment (rfranzke) [inline docs/usage/shoot_supported_architectures.md]:
```suggestion
If no value is specified for the architecture field, it defaults to `amd64`. For a valid shoot object, a machine type should be present in the respective `CloudProfile` with the same CPU architecture as specified in the `Shoot` yaml.
```

Don't we need a machine image with the right architecture as well?
Can you maybe also show some short YAML example of such `CloudProfile`?

File: docs/usage/shoot_supported_architectures.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+Currently, Gardener supports two most widely used CPU architectures:-
```
Comment (rfranzke) [inline docs/usage/shoot_supported_architectures.md]:
```suggestion
Currently, Gardener supports two most widely used CPU architectures:
```

**Review by rfranzke — CHANGES_REQUESTED** [review]

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**